Improve gRPC supported platforms doc (#21689)
dotnet/AspNetCore.Docs: aspnetcore/grpc/supported-platforms.md

By [James Newton-King](https://twitter.com/jamesnk)

This article discusses the requirements and supported platforms for using gRPC with .NET. There are different requirements for the two major gRPC tasks:

* Hosting gRPC services on the server.
* Calling gRPC services from a client.

gRPC takes advantage of advanced features available in HTTP/2. HTTP/2 isn't supported everywhere, but a second wire-format using HTTP/1.1 is available for gRPC:

* [`application/grpc`](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md) - gRPC over HTTP/2 is how gRPC is typically used.

> * [`application/grpc-web`](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md) - gRPC-Web modifies the gRPC protocol to be compatible with HTTP/1.1. gRPC-Web can be used in more places. gRPC-Web can be used by browser apps and it can be used in networks without complete support for HTTP/2. Two advanced gRPC features are no longer supported: client streaming and bidirectional streaming.

Reviewer: ```suggestion
* [`application/grpc-web`](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md) - gRPC-Web modifies the gRPC protocol to be compatible with HTTP/1.1. gRPC-Web can be used in more places. gRPC-Web can be used by browser apps and in networks without complete support for HTTP/2. Two advanced gRPC features are no longer supported: client streaming and bidirectional streaming.
```

> gRPC on .NET supports both wire-formats. gRPC over HTTP/2 is used by default. For information on setting up gRPC-Web, see <xref:grpc/browser>.

Reviewer: ```suggestion
gRPC on .NET supports HTTP 1.1 and HTTP/2 , HTTP/2 is used by default. For information on setting up gRPC-Web, see <xref:grpc/browser>.
```

Author: Updated change slightly. Would prefer to talk about wire format than HTTP version.

Reviewer: That's fine, only because you define it above.

> HTTP/2 isn't supported everywhere, but a second wire-format using HTTP/1.1 is available for gRPC:

It's common jargon that can be difficult for non-experts. I doubt it MTs.

## ASP.NET Core gRPC server requirements

Hosting gRPC services with ASP.NET Core requires .NET Core 3.x or later.

> [!div class="checklist"]
>
> * .NET 5 or later

> > * .NET Core 3

Reviewer: ```suggestion
> * .NET Core 3.1
```

Author: gRPC for .NET supports .NET Core 3

For information about configuring ASP.NET Core servers to run gRPC, see <xref:grpc/aspnetcore#server-options>.

### Azure services

> [!div class="checklist"]
>
> * [Azure Kubernetes Service (AKS)](https://azure.microsoft.com/services/kubernetes-service/)
> * [Azure App Service](https://azure.microsoft.com/services/app-service/)&dagger;

&dagger;Azure App Service doesn't support hosting gRPC over HTTP/2. gRPC-Web is a compatible alternative.

Work is in-progress to improve support for gRPC with HTTP/2 in Azure App Service. For more information, see [this GitHub issue](https://github.com/dotnet/AspNetCore/issues/9020).

## .NET gRPC client requirements

> The [Grpc.Net.Client](https://www.nuget.org/packages/Grpc.Net.Client/) package supports gRPC over HTTP/2 on .NET Core 3 and .NET 5 or later.

Reviewer: ```suggestion
The [Grpc.Net.Client](https://www.nuget.org/packages/Grpc.Net.Client/) package supports gRPC over HTTP/2 on .NET Core 3.1 and .NET 5 or later.
```

Limited support is available for gRPC over HTTP/2 on .NET Framework. Other .NET versions such as UWP, Xamarin and Unity don't have required HTTP/2 support, and must use gRPC-Web instead.

> [!div class="checklist"]
>
> * .NET 5 or later

> > * .NET Core 3

Reviewer: ```suggestion
> * .NET Core 3.1
```
